test/unit/modules.spec.ts

```diff
@@ -2,6 +2,86 @@ import * as tstl from "../../src";
 import { TSTLErrors } from "../../src/TSTLErrors";
 import * as util from "../util";
 
+describe("module import/export elision", () => {
+    const moduleDeclaration = `
+        declare module "module" {
+            export type Type = string;
+            export declare const value: string;
+        }
+    `;
+
+    const expectToElideImport = (code: string) => {
+        const lua = util.transpileString(
+            { "module.d.ts": moduleDeclaration, "main.ts": code },
+            undefined,
+            false,
+        );
+
+        expect(() => util.executeLua(lua)).not.toThrow();
+    };
+
+    test("should elide named type imports", () => {
+        expectToElideImport(`
+            import { Type } from "module";
+            const foo: Type = "bar";
+        `);
+    });
+
+    test("should elide named value imports used only as a type", () => {
+        expectToElideImport(`
+            import { value } from "module";
+            const foo: typeof value = "bar";
+        `);
+    });
+
+    test("should elide namespace imports with unused values", () => {
+        expectToElideImport(`
+            import * as module from "module";
+            const foo: module.Type = "bar";
+        `);
+    });
+
+    test("should elide type exports", () => {
+        const code = `
+            // TODO: Add some Lua Types to transpileStringResult
```

Comment:
Was this TODO meant for this PR?

Reply:
No, it's just a reminder to remove it once we'll have some global types there. Although it should error anyway if this symbol would be declared, so I'll remove it.